Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/checkins/TraceLocationCheckInRepository.kt

```diff
@@ -0,0 +1,28 @@
+package de.rki.coronawarnapp.eventregistration.checkins
+
+import de.rki.coronawarnapp.eventregistration.storage.TraceLocationDatabase
+import de.rki.coronawarnapp.eventregistration.storage.dao.TraceLocationCheckInDao
+import de.rki.coronawarnapp.eventregistration.storage.entity.TraceLocationCheckInEntity
+import kotlinx.coroutines.flow.Flow
+import javax.inject.Inject
+
+class TraceLocationCheckInRepository @Inject constructor(
+    traceLocationDatabaseFactory: TraceLocationDatabase.Factory
+) {
+
+    private val traceLocationDatabase: TraceLocationDatabase by lazy {
+        traceLocationDatabaseFactory.create()
+    }
+
+    private val traceLocationCheckInDao: TraceLocationCheckInDao by lazy {
+        traceLocationDatabase.eventCheckInDao()
+    }
+
+    val allCheckIns: Flow<List<TraceLocationCheckIn>> =
+        traceLocationCheckInDao
+            .allEntries()
+
+    suspend fun addCheckIn(checkIn: TraceLocationCheckIn) =
+        traceLocationCheckInDao
+            .insert(checkIn as TraceLocationCheckInEntity)
```

Comment:
this is dangerous, if the interface is implemented by another class , this will crash. I don't see the benefit of the interface here

Reply:
I also think that the database entity model doesn't need to implement the `TraceLocationCheckIn` interface, only the models that interact with other components of the app. (I am also not doing this in the TraceLocation Repo).